core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -0,0 +1,736 @@
+                if (classPath != null && !classPath.trim().isEmpty()) {
+                    ClassLoader classLoader = new ChildFirstClassLoader(classPath, this.getClass().getClassLoader());
+                    RemoteLogMetadataManager delegate = createDelegate(classLoader, rlmConfig.remoteLogMetadataManagerClassName());
+                    return new ClassLoaderAwareRemoteLogMetadataManager(delegate, classLoader);
+                } else {
+                    return createDelegate(this.getClass().getClassLoader(), rlmConfig.remoteLogMetadataManagerClassName());
+                }
+            }
+        });
+    }
+
+    private void configureRLMM() {
+        final Map<String, Object> rlmmProps = new HashMap<>(rlmConfig.remoteLogMetadataManagerProps());
+
+        rlmmProps.put(KafkaConfig.BrokerIdProp(), brokerId);
+        rlmmProps.put(KafkaConfig.LogDirProp(), logDir);
+        remoteLogMetadataManager.configure(rlmmProps);
+    }
+
+    public void startup() {
+        // Initialize and configure RSM and RLMM. This will start RSM, RLMM resources which may need to start resources
+        // in connecting to the brokers or remote storages.
+        configureRSM();
+        configureRLMM();
+    }
+
+    public RemoteStorageManager storageManager() {
+        return remoteLogStorageManager;
+    }
+
+    private Stream<Partition> filterPartitions(Set<Partition> partitions) {
+        // We are not specifically checking for internal topics etc here as `log.remoteLogEnabled()` already handles that.
+        return partitions.stream().filter(partition -> partition.log().exists(UnifiedLog::remoteLogEnabled));
+    }
+
+    private void cacheTopicPartitionIds(TopicIdPartition topicIdPartition) {
+        Uuid previousTopicId = topicPartitionIds.put(topicIdPartition.topicPartition(), topicIdPartition.topicId());
+        if (previousTopicId != null && previousTopicId != topicIdPartition.topicId()) {
+            LOGGER.info("Previous cached topic id {} for {} does not match updated topic id {}",
+                    previousTopicId, topicIdPartition.topicPartition(), topicIdPartition.topicId());
+        }
+    }
+
+    // for testing
+    public RLMScheduledThreadPool rlmScheduledThreadPool() {
+        return rlmScheduledThreadPool;
+    }
+
+    /**
+     * Callback to receive any leadership changes for the topic partitions assigned to this broker. If there are no
+     * existing tasks for a given topic partition then it will assign new leader or follower task else it will convert the
+     * task to respective target state(leader or follower).
+     *
+     * @param partitionsBecomeLeader   partitions that have become leaders on this broker.
+     * @param partitionsBecomeFollower partitions that have become followers on this broker.
+     * @param topicIds                 topic name to topic id mappings.
+     */
+    public void onLeadershipChange(Set<Partition> partitionsBecomeLeader,
```

Comment:
I didn't see any place we invoke this method. I think `ReplicaManager#becomeLeaderOrFollower` should be responsible to invoke this method, right?